### ISSUE TITLE
Automatically validate location data of `POI` model

### ISSUE THREAD

**Issue by timoludwig** [Integreat/integreat-cms#1000]
### Motivation

At the moment, it is possible to insert incorrect data for the exact position of `POI`s. Since there are many possible pitfalls when creating `POI`s, we should validate the input and auto-fill as much as possible.
And since the map feature of the app breaks when we deliver incorrect data, this should ideally be implemented before the initial production release.

### Proposed Solution

Possible ideas (check effort vs. usability on each step):

- [ ] Validate whether an address format is correct
- [ ] Validate whether a street exists in the given city
- [ ] Suggest available streets on the selected city
- [ ] Auto-fill city after zip code has been inserted
- [ ] Auto-fill coordinates after address is complete
- [ ] If coordinates are manually changed after auto-fill, check for consistency with the address data and raise a warning if the positions are completely off

Check which data sources and APIs can be used for this task.

**Comment by svenseeberg** [comment]
Honestly, I do not think that this is that important. As with all content that is created in the app, the content creators need to take care that everything is "correct". I think it would be most important to show a preview of the location on a small embedded map to simply show where the location actually is on a map. I would also argue that this is not necessary for a release.

**Comment by ulliholtgrave** [comment]
I agree with Sven, a validation is surely a nice thing to have, but the wordpress doesn't have it. Therefore I suggest to postpone it to Version 2.0 and add the prio:high label.

**Comment by timoludwig** [comment]
Alright, seems reasonable! :+1:

**Comment by timoludwig** [comment]
We could use our geocoding service for this:
http://nominatim.maps.tuerantuer.org/nominatim/search.php?street=Am%20B%C3%BChl&postalcode=86199

API documentation: https://nominatim.org/release-docs/develop/api/Overview/

**Comment by dkehne** [comment]
Isn't that also related to #1035 ?

**Comment by timoludwig** [comment]
> Isn't that also related to #1035 ?

Yes, it's related, but can be implemented independently of each other.

**Comment by timoludwig** [comment]
Since this is now mostly done, I decreased the prio of the remaining tasks and moved it back into the backlog

**Comment by JoeyStk** [comment]
@laravoelk could you please ask in the team whether or not the last two items from the list should still be implemented. The two items are "Suggest available streets on the selected city" and "If coordinates are manually changed after auto-fill, check for consistency with the address data and raise a warning if the positions are completely off". For both the effort on our side would be rather high, so please check if the use justifies the effort :) Thank you in advance :) If you have any questions about this, please don't hesitate to text us :)

**Comment by laravoelk** [comment]
To be honest, I do not think that this is really needed. I have not heard about any requests going into this direction and also if the effort from your side is pretty high, I guess it is not worth it. :-)
